### PR TITLE
first edition

### PR DESCRIPTION
**Pull request by ws752499660**

### CODE REVIEW

File: mcr.cpp [hunk 1]
```diff
@@ -1,46 +1,51 @@
...
-            cout << "Player 2: "; 
-        cout << "Which cell to mark? i:[1..3], j:[1..3]: "; 
+    int count=0;
+    for (int n=0; n<9; n++){
+        turn = !turn; // use the not-operator to change true to false or false to true.
+        if (turn == false)
+            cout << "Player 1: ";
+        else
+        cout << "Player 2: ";
+        cout << "Which cell to mark? i:[0..2], j:[0..2]: ";
```
Comment (ws752499660) [inline mcr.cpp:28]: the output in last edition code is wrong.

File: mcr.cpp [hunk 2]
```diff
@@ -1,46 +1,51 @@
...
-    bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first. 
+int main(){
+    int i, j;
+    char game[3][3] = {' '}; // Tic-tac-toe
+    char player1 = 'X';
+    char player2 = 'O';
+    bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first.
     cout << "X = Player 1" << endl << "O = Player 2" << endl;
-    for (int n=0; n<9; n++){ 
-        turn = !turn;  // use the not-operator to change true to false or false to true.
```
Comment (kityanho3-c) [inline mcr.cpp:23]: suggest to move this line to the bottom of the for loop, to make it consistent with the comment in line 20

File: mcr.cpp [hunk 3]
```diff
@@ -1,46 +1,51 @@
...
         cin >> i >> j;
-        if (turn == false) 
-            game[i][j] = player1; 
-        else  
-            game[i][j] = player2; 
-        if (isWin(game)){ 
+        if(i>2 || i<0 || j>2 || j<0){
+            cout<<"Input error! Please input i and j again";
+            continue;
+        }
```
Comment (ws752499660) [inline mcr.cpp:33]: add the check of input of i and j

File: mcr.cpp [hunk 4]
```diff
@@ -1,46 +1,51 @@
...
+    bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first.
     cout << "X = Player 1" << endl << "O = Player 2" << endl;
-    for (int n=0; n<9; n++){ 
-        turn = !turn;  // use the not-operator to change true to false or false to true. 
-        if (turn == false) 
-            cout << "Player 1: "; 
-        Else 
-            cout << "Player 2: "; 
-        cout << "Which cell to mark? i:[1..3], j:[1..3]: "; 
+    int count=0;
```
Comment (harreasonsolo) [inline mcr.cpp:22]: add the var of count

File: mcr.cpp [hunk 5]
```diff
@@ -1,46 +1,51 @@
...
-    char game[3][3] = {' '}; // Tic-tac-toe 
-    char player1 = 'X'; 
-    char player2 = 'O'; 
-    bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first. 
+int main(){
+    int i, j;
+    char game[3][3] = {' '}; // Tic-tac-toe
+    char player1 = 'X';
+    char player2 = 'O';
+    bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first.
```
Comment (ws752499660) [inline mcr.cpp]: maybe the trun should be false for the palyer1 firstly.

File: mcr.cpp [hunk 6]
```diff
@@ -1,46 +1,51 @@
...
-    for (int n=0; n<9; n++){ 
-        turn = !turn;  // use the not-operator to change true to false or false to true. 
-        if (turn == false) 
-            cout << "Player 1: "; 
-        Else 
-            cout << "Player 2: "; 
-        cout << "Which cell to mark? i:[1..3], j:[1..3]: "; 
+    int count=0;
+    for (int n=0; n<9; n++){
+        turn = !turn; // use the not-operator to change true to false or false to true.
```
Comment (ws752499660) [inline mcr.cpp]: this should be moved to the line after the player operate

File: mcr.cpp [hunk 7]
```diff
@@ -1,46 +1,51 @@
...
-    // show the game to console 
-    cout << game[0][0] << " " << game[0][1] << " " << game[0][2] << endl; 
-    cout << game[1][0] << " " << game[1][1] << " " << game[1][2] << endl; 
-    cout << game[2][0] << " " << game[2][1] << " " << game[2][2] << endl; 
-    return 0; 
-} 
+            break; // need to terminate the problem
+        }
+    }
+    if (count==8) // all celles with i=0 to 2 have been inputted above but no winner yet
```
Comment (harreasonsolo) [inline mcr.cpp:44]: to end the game with a "Tie" when all cells is full.

**Review by ws752499660 — COMMENTED** [review]
frist change

**Pull request merged**